src/overlays/ovl_i5/fox_ti.c

```diff
@@ -1,138 +1,5375 @@
+                        actor->fwork[0] = (RAND_FLOAT(5.0f) + 5.0f) * (f32) var_s1;
+
+                        if (Rand_ZeroOne() < 0.5f) {
+                            var_s1 = -1;
+                        } else {
+                            var_s1 = 1;
+                        }
+                        actor->fwork[1] = (RAND_FLOAT(5.0f) + 5.0f) * (f32) var_s1;
+                        Matrix_Push(&gCalcMatrix);
+                        Matrix_RotateY(gCalcMatrix, RAND_FLOAT(360.0f) * M_DTOR, 0);
+                        Matrix_MultVec3f(gCalcMatrix, &D_i5_801B8E44, &sp60);
+                        Matrix_Pop(&gCalcMatrix);
+                        actor->vel.x = sp60.x;
+                        var_s1 = (D_i5_801BBF00[i].unk_00.pos.y > 250.0f)    ? 1
+                                 : (D_i5_801BBF00[i].unk_00.pos.y == 250.0f) ? 0
+                                                                             : -1;
+                        actor->vel.y = ((f32) var_s1 - RAND_FLOAT(0.5f)) + 0.5f;
+                        actor->vel.z = (boss->vel.z * 0.5f) + sp60.z;
+                        actor->gravity = 0.1f;
+                    }
+                }
+                D_i5_801BBF00[i].unk_26 = 4;
+            }
+        }
+        boss->unk_044 = 1;
+    } else {
+        boss->obj.pos.z = gPlayer[0].pos.z;
+    }
+    // FAKE
+    var_s1 = boss->timer_050;
+    switch (var_s1) {
+        case 100:
+            actor = func_800A3608(OBJ_ACTOR_189);
+            if (actor != NULL) {
+                if (Rand_ZeroOne() < 0.5f) {
+                    var_s1 = -1;
+                } else {
+                    var_s1 = 1;
+                }
+                actor->fwork[0] = (Rand_ZeroOne() + 1.0f) * (f32) var_s1;
+                if (Rand_ZeroOne() < 0.5f) {
+                    var_s1 = -1;
+                } else {
+                    var_s1 = 1;
+                }
+
+                actor->fwork[1] = (Rand_ZeroOne() + 1.0f) * (f32) var_s1;
+                actor->state = 0x28;
+                actor->unk_046 = 25;
+                actor->unk_048 = 0;
+                actor->unk_04A = 6;
+                actor->fwork[3] = 0.0f;
+                actor->fwork[4] = -200.0f;
+                actor->obj.pos.x = gPlayer[0].pos.x;
+                actor->obj.pos.y = 500.0f;
+                actor->obj.pos.z = gPlayer[0].unk_138 + actor->fwork[4];
+                actor->obj.rot.y = (RAND_FLOAT(5.0f) + 90.0f) - 2.5f;
+                actor->obj.rot.z = (RAND_FLOAT(5.0f) + 180.0f) - 2.5f;
+                actor->vel.y = -10.0f;
+                actor->gravity = 0.8f;
+            }
+            break;
+        case 120:
+            actor = func_800A3608(OBJ_ACTOR_189);
+            if (actor != NULL) {
+                if (Rand_ZeroOne() < 0.5f) {
+                    var_s1 = -1;
+                } else {
+                    var_s1 = 1;
+                }
+                actor->fwork[0] = (RAND_FLOAT(5.0f) + 5.0f) * (f32) var_s1;
```

Comment:
```suggestion
                actor->fwork[0] = (RAND_FLOAT(5.0f) + 5.0f) * ((Rand_ZeroOne() < 0.5f) ? 1 : -1);
```
You don't have to make this change, but I'm curious if something like this still matches. If so, macroing up `((Rand_ZeroOne() < 0.5f) ? 1 : -1)` could be reasonable.

Reply:
forgot to use that on this function ah, got a non-fake match now thanks